contribute/design/1.x/csi/20190805-csi-zfspv-volume-provisioning.md

```diff
@@ -261,6 +258,123 @@ When CSI will get volume destroy request, it will destroy the created zvol and a
 ### 4. CSI ZFSPV property change
 There will be a watcher watching for this ZFSVolume custom resource in the agent. We can update the ZFSVolume custom resource with the desired property and the watcher of this custom resource will apply the changes to the corresponding volume.
 
+### 5. CSI SnapShot
+
```

Comment:
Could you add a note on why this approach of CR was selected over - possibly having the CSI controller directly invoking the snapshot command on the corresponding CSI node-agent.

Reply:
added, Please check.